src/humancellatlas/data/metadata/api.py

```diff
@@ -306,11 +314,32 @@ def __init__(self,
         self.publications = set(ProjectPublication.from_json(publication)
                                 for publication in content.get('publications', []))
         self.contributors = {ProjectContact.from_json(contributor) for contributor in content.get('contributors', [])}
-        self.insdc_project_accessions = set(content.get('insdc_project_accessions', []))
-        self.geo_series_accessions = set(content.get('geo_series_accessions', []))
-        self.array_express_accessions = set(content.get('array_express_accessions', []))
-        self.insdc_study_accessions = set(content.get('insdc_study_accessions', []))
         self.supplementary_links = set(content.get('supplementary_links', []))
+        self.accessions = []
+        for a in content.keys():
+            if a.endswith('_accessions'):
+                type_ = a.split('_accessions')[0]
```

Comment:
```suggestion
                type_ = key[:-len('_accessions')]
```
Different way of doing the same thing, however this way ensures we are only removing from the end of the string. Consider something like `'foo_accessions_bar_accessions'.split('_accessions')`

Reply:
partition and rpartition only split once